### ISSUE TITLE
Unable to find a matching set of capabilities with Firefox

### ISSUE THREAD

**Issue by ylavoie** [teodesian/Selenium-Remote-Driver#375]
Using S::R::D 1.25+, Geckodriver 1.19.1 and Firefox 57.0.4, the only way I was able to get a connection was to remove the `platform => 'ANY'` in the Driver->new.
It is documented that there is a problem in newer Selenium versions with platform negociation.
I suggest to remove it when FORCE_WD3 is effective for Firefox.

**Comment by teodesian** [comment]
Is this when using Selenium::Firefox or S::R::D?  I can figure it out if you don't tell me but you could save me some time by letting me know.

**Comment by ylavoie** [comment]
Sorry. I used Selenium::Firefox like this:
```
    require Selenium::Firefox::Profile;
    my $caps = {
        extra_capabilities => {
            'moz:firefoxOptions' => {
              args    => [ '-headless' ],
            },
        },
    };
    $Selenium::Remote::Driver::FORCE_WD3 = 1;
    $driver = Selenium::Firefox->new(%{$caps});
```

**Comment by teodesian** [comment]
Thanks.  I'll look at a fix for this tomorrow morning.

**Comment by teodesian** [comment]
I think @ylavoie's patch fixes this -- I'm going to test and close out tomorrow morning.

**Comment by teodesian** [comment]
Without your code I merged, I cannot reproduce the problem with your provided script.  Perhaps there was something else going on?

**Comment by teodesian** [comment]
This should be fixed in #396, we now forcibly erase the platform, version and browser name from the caps passed.
